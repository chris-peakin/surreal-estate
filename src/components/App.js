import "../styles/App.css";
import React, { Link, Route, Switch } from 'react';
// import Switch from 'react-router-dom';
import NavBar from './NavBarComponent';
import AddProperty from "./AddPropertyComponent";

function App() {
  return (
    <>
      <h2>Surreal Estate</h2>
      <NavBar>
        <Link className="item" to="/">
          View Properties
        </Link>
        <Link className="item" to="add-property">
          Add Properties
        </Link>
      </NavBar>
      <Switch>
        <Route exact path="/" />
        <Route exact path="/add-property" component={Properties}>
          <AddProperty />
        </Route>
      </Switch>
    </>
  );
}

export default App;
