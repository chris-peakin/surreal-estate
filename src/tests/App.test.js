import { render } from "@testing-library/react";
import App from "../components/App";

test("renders the app correctly", () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot;
});
