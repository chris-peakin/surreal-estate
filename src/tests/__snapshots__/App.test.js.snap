// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the app correctly 1`] = `
Array [
  <div
    className="navbar"
  >
    <img
      alt="estate-logo"
      className="logo"
      src="https://mcrcodes.s3.eu-west-2.amazonaws.com/course/surreal-estate/logo.png"
    />
    <ul
      className="navbar-links"
    >
      <li
        className="navbar-links-item"
      >
        <a
          className="navbar-links-link"
          href="/"
          onClick={[Function]}
        >
          View Properties
        </a>
      </li>
      <li
        className="navbar-links-item"
      >
        <a
          className="navbar-links-link"
          href="/add-property"
          onClick={[Function]}
        >
          Add a Property
        </a>
      </li>
      <li
        className="navbar-links-item"
      >
        <a
          className="navbar-links-link"
          href="/saved-properties"
          onClick={[Function]}
        >
          Favourite Properties
        </a>
      </li>
    </ul>
    <div
      className="login-button"
    >
      <span
        style={
          Object {
            "transition": "opacity 0.5s",
          }
        }
      >
        <button
          className="kep-login-facebook metro"
          onClick={[Function]}
          type="button"
        >
          Login
        </button>
        <style
          dangerouslySetInnerHTML={
            Object {
              "__html": Array [
                Array [
                  9,
                  ".kep-login-facebook{font-family:Helvetica,sans-serif;font-weight:700;-webkit-font-smoothing:antialiased;color:#fff;cursor:pointer;display:inline-block;font-size:calc(.27548vw + 12.71074px);text-decoration:none;text-transform:uppercase;transition:background-color .3s,border-color .3s;background-color:#4c69ba;border:calc(.06887vw + .67769px) solid #4c69ba;padding:calc(.34435vw + 13.38843px) calc(.34435vw + 18.38843px)}.kep-login-facebook.small{padding:calc(.34435vw + 3.38843px) calc(.34435vw + 8.38843px)}.kep-login-facebook.medium{padding:calc(.34435vw + 8.38843px) calc(.34435vw + 13.38843px)}.kep-login-facebook.metro{border-radius:0}.kep-login-facebook .fa{margin-right:calc(.34435vw + 3.38843px)}",
                  "",
                ],
              ],
            }
          }
        />
      </span>
    </div>
  </div>,
  <div />,
  <div>
    <div
      className="sidebar"
    >
      <form
        onSubmit={[Function]}
      >
        <input
          onChange={[Function]}
          placeholder="Property title search..."
          type="text"
          value=""
        />
        <button
          type="submit"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505.04 442.66l-99.71-99.69c-4.5-4.5-10.6-7-17-7h-16.3c27.6-35.3 44-79.69 44-127.99C416.03 93.09 322.92 0 208.02 0S0 93.09 0 207.98s93.11 207.98 208.02 207.98c48.3 0 92.71-16.4 128.01-44v16.3c0 6.4 2.5 12.5 7 17l99.71 99.69c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.59.1-33.99zm-297.02-90.7c-79.54 0-144-64.34-144-143.98 0-79.53 64.35-143.98 144-143.98 79.54 0 144 64.34 144 143.98 0 79.53-64.35 143.98-144 143.98zm.02-239.96c-40.78 0-73.84 33.05-73.84 73.83 0 32.96 48.26 93.05 66.75 114.86a9.24 9.24 0 0 0 14.18 0c18.49-21.81 66.75-81.89 66.75-114.86 0-40.78-33.06-73.83-73.84-73.83zm0 96c-13.26 0-24-10.75-24-24 0-13.26 10.75-24 24-24s24 10.74 24 24c0 13.25-10.75 24-24 24z"
            />
          </svg>
        </button>
      </form>
      <div>
        Sort By City:
        <a
          href="/?query={\\"city\\":\\"Manchester\\"}"
          onClick={[Function]}
        >
          Manchester
        </a>
        <a
          href="/?query={\\"city\\":\\"Leeds\\"}"
          onClick={[Function]}
        >
          Leeds
        </a>
        <a
          href="/?query={\\"city\\":\\"Sheffield\\"}"
          onClick={[Function]}
        >
          Sheffield
        </a>
        <a
          href="/?query={\\"city\\":\\"Liverpool\\"}"
          onClick={[Function]}
        >
          Liverpool
        </a>
      </div>
      <div>
        Sort By Price:
        <a
          href="/?sort={\\"price\\":1}"
          onClick={[Function]}
        >
          Ascending
        </a>
        <a
          href="/?sort={\\"price\\":-1}"
          onClick={[Function]}
        >
          Descending
        </a>
      </div>
    </div>
  </div>,
  <div
    className="individual_properties"
  />,
]
`;
